### PR TITLE
Add information about subscriptions and agent-backend interaction to architecture overview

### LINKED ISSUE #2602: Add text to explain and distinguish Sensu
### Description
Draft from Vikas (reviewed by Sean) to add to architecture overviews in docs:

Sensu works slightly differently from other monitoring solutions - instead of provisioning each device or container to be monitored in the Sensu backend, you just create ‘subscriptions’ in the backend (e.g. webserver). Then you define what checks need to be done for all devices that are part of this subscription. You do NOT need to list each server that needs to be monitored in the backend.

Then you install lightweight Sensu agents on all the servers (or container sidecars) which you want to monitor. When starting the agent, you only need to specify the IP address of the backend server and the list of subscriptions the agent belongs to. The required files for running the checks are automatically downloaded from an asset repository (e.g. bonsai.sensu.io or another local one) and the checks are scheduled on each agent.

When the agent connects to the backend, it is automatically registered without requiring any further provisioning and the agent will execute the checks sent to its subscriptions by the backend. 

The backend keeps track of ‘self-registered’ agents  and if it loses a keepalive signal from any of them, it is flagged as down and an event is generated. If a server (agent) shuts down gracefully, the agent is automatically de-registered from the backend and no alert is generated (this behavior can be easily configured if desired).

agent.yml 

backend_url
subscriptions: [“mysql”]

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
https://docs.sensu.io/sensu-go/latest/operations/deploy-sensu/install-sensu/#architecture-overview
https://docs.sensu.io/sensu-go/latest/operations/deploy-sensu/deployment-architecture/

### Expected behavior
Our docs include info to help new users differentiate Sensu from other products that they may be familiar with (including Sensu concepts like self configuration, self registering agents).

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Adds distinguishing information about Sensu, subscription model, and agent-backend interaction in https://docs.sensu.io/sensu-go/latest/operations/deploy-sensu/install-sensu/#architecture-overview

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2602

### CODE REVIEW

**Comment by hillaryfraley** [comment]
@portertech I believe you already reviewed the source content (from Vikas) for this PR, but I did some reworking to combine it with the explanation we already had in the docs. So I thought it was worth requesting a second look before I publish it in these docs.

File: content/sensu-go/5.21/operations/deploy-sensu/install-sensu.md [hunk 1]
```diff
@@ -26,19 +26,31 @@ See [supported platforms][5] for more information.
...
 
+Sensu works differently from other monitoring and observability solutions.
+
+The **Sensu backend** is powered by an an embedded transport and [etcd][16] datastore.
+You do not need to provision each device, server, container, or sidecar you want to monitor or list each of them in the Sensu backend.
+Instead, you create [subscriptions][41] and define which checks to run for all infrastructure components that belong to each subscription.
+
+Then, you install **Sensu agents**: lightweight clients that run on the infrastructure components you want to monitor.
+Agents automatically register with Sensu as entities when you start them up and connect to the Sensu backend with no need for further provisioning.
+You only need to specify the IP address for the Sensu backend server and the subscriptions the agents belong to.
```
Comment (portertech) [inline content/sensu-go/5.21/operations/deploy-sensu/install-sensu.md]: With 6.0, you technically don't have to set `subscriptions` up front in agent.yml, you can add them to the agent's entity via the API later. Not sure if we want to "muddy the waters" here.

Comment (hillaryfraley) [inline content/sensu-go/5.21/operations/deploy-sensu/install-sensu.md:37]:
```suggestion
You only need to specify the IP address for the Sensu backend server.
```

Comment (hillaryfraley) [reply content/sensu-go/5.21/operations/deploy-sensu/install-sensu.md]: Thank you -- I'll update that for 6.0 and commit that change to this PR.

**Comment by hillaryfraley** [comment]
Thank you, @portertech ! I updated the 6.0 doc and propagated to all versions. Wdyt?

File: content/sensu-go/5.19/operations/deploy-sensu/install-sensu.md [hunk 2]
```diff
@@ -26,19 +26,31 @@ See [supported platforms][5] for more information.
 
 ## Architecture overview
 
+Sensu works differently from other monitoring and observability solutions.
+
+The **Sensu backend** is powered by an an embedded transport and [etcd][16] datastore.
+You do not need to provision each device, server, container, or sidecar you want to monitor or list each of them in the Sensu backend.
+Instead, you create [subscriptions][41] and define which checks to run for all infrastructure components that belong to each subscription.
```
Comment (portertech) [inline content/sensu-go/5.19/operations/deploy-sensu/install-sensu.md]: Subscriptions aren’t a real resource in Sensu. A User creates checks that target one or more subscriptions. Agents subscribe to subscriptions that align with their host’s responsibilities/roles.

**Pull request merged**